pkg/snapshotter/snapshotter_csi.go

```diff
@@ -601,7 +601,26 @@ func (c *csiDriver) RestoreVolumeClaim(opts ...Option) (*v1.PersistentVolumeClai
 				logrus.Debugf("setting size of pvc %s/%s same as snapshot size %s", pvc.Namespace, pvc.Name, quantity.String())
 				pvc.Spec.Resources.Requests[v1.ResourceStorage] = quantity
 			}
-
+		}
+		// PB-8482 if restore size is nil then set the pvc size to annotation restoreSize
+		if snapshot.Status.RestoreSize == nil {
+			restoreSizeStr, ok := snapshot.Annotations["restoreSize"]
+			if !ok {
+				return nil, fmt.Errorf("restoreSize annotation not found")
+			}
+			restoreSize, err := resource.ParseQuantity(restoreSizeStr)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse restoreSize: %v", err)
+			}
+			// Update the pvc size only when the volumesnapshot size is greater than PVC size
+			if restoreSize.Cmp(pvc.Spec.Resources.Requests[v1.ResourceStorage]) == 1 {
```

Comment:
For KDMP backup we found that restore failed even when the snapshot size was less than PVC size. Most of the storage provider allows upto 80% of the PVC size and sometimes filesystem takes up some size for metadata. In such cases restore will fail if snapshot size is less than PVC but greater. 
The fix we made was to check if snapshot size is less 75% of PVC size. Please see if this is applicable for CSI also

Reply:
`		// Below two cases, we will change the PVC original size.
		// case1: If the snapshot size is greater than the PVC size. In some of the filesystem
		// provisioned PVC, the content of the PVC is allowed to grow beyond the PVC size.
		// case2: If the snapshot size + 10% of the snapshot size goes beyind the PVC size.
		// We also allow the customer to configure the sizePercentage to configured, if the 10% does not
		// satisfy their storage provisioner.
		// KDMP_RESTORE_PVC_SIZE_PERCENTAGE parameter need to be added to the kdmp-config configmap in kube-system namespace.
		sizePercentage := utils.GetRestorePvcSizePercentage()
		if sizePercentage == 0 {
			sizePercentage = defaultPvcSizeIncreasePercentage
		}
		logrus.Infof("StartRestore: sizePercentage is - %v", sizePercentage)
		totalSize := int64(bkpvInfo.TotalSize)
		newPvcQuantity := resource.NewQuantity(
			totalSize+((int64(sizePercentage)*int64(bkpvInfo.TotalSize))/int64(100)),
			resource.BinarySI,
		)
		pvcQuantity := pvc.Spec.Resources.Requests[v1.ResourceStorage]
		if (pvcQuantity.Cmp(*newPvcQuantity)) == -1 {
			logrus.Debugf("StartRestore: setting size of pvc %s/%s to accommodate snapshot size with buffer %s", pvc.Namespace, pvc.Name, newPvcQuantity.String())
			pvc.Spec.Resources.Requests[v1.ResourceStorage] = *newPvcQuantity
		}`
		
		Above is the code put up for KDMP and it csi i didn't find it and couldn't replicate too. Most probably that's why variable name was KDMP_RESTORE_PVC_SIZE_PERCENTAGE and not related to CSI

Reply:
Yes we noticed it specifically in KDMP so this was implemented with configurable values through kdmp-config. I agree this may not be the case for CSI. If the scenario is tested for CSI also then great.